include/libpmemobj++/experimental/basic_string.hpp

```diff
@@ -2254,6 +2254,11 @@ basic_string<CharT, Traits>::initialize(Args &&... args)
 {
 	assert(pmemobj_tx_stage() == TX_STAGE_WORK);
 
+	/* Mark sso array as initialized */
+#if LIBPMEMOBJ_CPP_VG_MEMCHECK_ENABLED
+	VALGRIND_MAKE_MEM_DEFINED(&sso.data, sizeof(sso.data));
+#endif
+
```

Comment:
This looks wrong. This function initializes the object, so it doesn't matter whether the previous content was initialized or not.